packages/tools/create-kadena-app/templates/angular/src/app/app.service.ts

```diff
@@ -0,0 +1,60 @@
+import { Injectable } from '@angular/core';
+import { Pact, signWithChainweaver } from '@kadena/client'
+import { environment } from './../environments/environment';
+
+const { kadenaNetworkId, kadenaChainId, kadenaHost } = environment;
+const apiHost = `https://${kadenaHost}/chainweb/0.0/${kadenaNetworkId}/chain/${kadenaChainId}/pact`
+
+const accountKey = (account: string): string => account.split(':')[1]
+
+@Injectable({ providedIn: 'root' })
+export class AppService {
+
+  async writeMessage(account:string, message:string) {
+    try {
+      const transactionBuilder = Pact.modules['free.cka-message-store']['write-message'](account, message)
+      .addCap('coin.GAS', accountKey(account))
+      .addCap('free.cka-message-store.ACCOUNT-OWNER' , accountKey(account), account)
+      .setMeta({
+        ttl: 28000,
+        gasLimit: 100000,
+        chainId: kadenaChainId,
+        gasPrice: 0.000001,
+        sender: account,
+      }, kadenaNetworkId)
+
+      await signWithChainweaver(transactionBuilder)
+
```

Comment:
Should we add a comment indicating that this is changed by reference?
```suggestion

      // `signWithChainweaver` modifies the `transactionBuilder` that's passed, 
      //   so we use this further down
      // It also returns an array with the transactionBuilders that are passed 
      //   (if one is passed it'll be an array of 1)

      await signWithChainweaver(transactionBuilder)

```